types/src/core/data_source.ts

```diff
@@ -1,12 +1,26 @@
 export type QdrantCluster = "cluster-0";
 export const DEFAULT_QDRANT_CLUSTER: QdrantCluster = "cluster-0";
 
+interface EmbedderType {
+  provider_id: string;
+  model_id: string;
+  splitter_id: string;
+}
+
+interface EmbedderConfigType {
+  embedder: EmbedderType;
+}
+
 export type CoreAPIDataSourceConfig = {
+  // TODO(2024-05-27 flav) Remove once migrated to embedder.
   provider_id: string;
   model_id: string;
+  splitter_id: string;
+
+  embedder_config: EmbedderConfigType;
+
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   extras?: any | null;
```

Comment:
We likely want to remove it from here in that PR?

Reply:
Done.